File: scripts/lex.lua
profile = "unimrcp:uni2"
promptCmd = "say: Welcome to Booking service by Connector73. How can I help you?"
-- Compose recognition command
recogCmd = " detect:" .. profile .. " {start-input-timers=false,no-input-timeout=5000}builtin:speech/transcribe"
-- Compose final message
finalMessage = "Thank you. See you next time"
-- constansts
asrtag = "names"

function onInput(s, type, obj)
    	freeswitch.consoleLog("INFO", "BOT Callback with type " .. type .. "\n");
	return "true"
end

freeswitch.consoleLog("INFO", "BOT Started\n")
-- Sleep a little bit to give media time to be fully up.
session:sleep(500)
session:set_tts_params(profile, "Salli");
session:setInputCallback("onInput");

tryagain = 1
-- Magic happens here.
while (session:ready() == true and tryagain == 1) do
        local command = promptCmd .. recogCmd
        freeswitch.consoleLog("INFO", "Command -> " .. command .."\n")
        session:execute("play_and_detect_speech", command)
        local text = session:getVariable('detect_speech_result')
        if (text ~= nil) then
                freeswitch.consoleLog("INFO", "XML <- " .. text .. "\n")
                if string.find(text, "xml") ~= nil then
                        local xml = require("xmlSimple").newParser()
                        local parsedXml = xml:ParseXmlText(text)
                        local dialogstate = parsedXml.result.interpretation.instance.dialogstate:value()
                        freeswitch.consoleLog("INFO", "BOT dialog state: " .. dialogstate .. "\n")
                        if (dialogstate == "ReadyForFulfillment") then
                                freeswitch.consoleLog("INFO", "Ready for fulfillment\n")
                                promptCmd = finalMessage
                                tryagain = 0
                        else
                                local result = parsedXml.result.interpretation.instance.message:value()
                                if (result == nil) then
                                        freeswitch.consoleLog("CRIT","Result is 'nil'\n")
                                        promptCmd = "say: I dont' understand you"
                                else
                                        promptCmd = "say: " .. result
                                end
				tryagain = 1
                        end
                end
        end
end
freeswitch.consoleLog("INFO", "Exit from the BOT\n")
session:speak(promptCmd)
session:sleep(1000)
session:hangup()

File: scripts/lex.js

// Profile to use
var profile = "unimrcp:uni2";

// Compose initial prompt command
var promptCmd = "say: Welcome to Amazon Lex. How can I help you?";

// Compose recognition command
var recogCmd = " detect:" + profile + " {start-input-timers=false,define-grammar=false,no-input-timeout=10000}builtin:speech/transcribe";

// Compose final message
var finalMessage = "Thank you. See you next time";

// Main recognition loop
while (session.ready()) {
        var command = promptCmd + recogCmd;

        // Play and detect speech
        session.execute("play_and_detect_speech", command);

        // Retrieve result
        var jsonResult = session.getVariable('detect_speech_result');
        if (typeof jsonResult != "undefined") {
                if (jsonResult.indexOf("Completion-Cause:") == -1) {

                        // Parse JSON result
                        var result = JSON.parse(jsonResult);
                        if (typeof result != "undefined") {
                                if (typeof result.transcript != "undefined") {
                                        // Compose next prompt command
                                        promptCmd = "say: " + result.message;
                                }

                                // Check dialog state
                                var state = result.dialogstate;

                                if (typeof state != "undefined") {

                                        if (state == "ReadyForFulfillment") {

                                                break;
                                        }
                                }
                        }
                }
                else {
                        consoleLog("INFO", "Recognition completed abnormally!\n");
                }
        }
        else {
                consoleLog("INFO", "No result!\n");
                break;
        }
}
 
// Speak the final message
session.speak(profile, "", finalMessage);
